### DISCUSSION TITLE
Mechanism for banning attack assets

### DISCUSSION THREAD

**Discussion by aggre** [dev-protocol/DIPs#43] [Ideas]
I still need to consider a lot, but I'll share a draft on mechanisms for curbing attack assets. What do you think, guys?

- Malicious assets are called "attack assets." It is an asset disguised for the purpose of spoofing, fraud, etc.
- Users can report to the protocol when they think an asset is an "attack asset." The protocol requests the reporter to deposit some DEV.
- Upon the report, the protocol will set randomly selected creators (within the same Market) as the jury.
- The jury will determine if the asset is an attack asset.
- When the asset is an attack asset, all rewards for the attack asset will be forfeited. Some will be re-allocated to the jury, and forfeited reward will burn some.
- If the reported asset's creator disagrees with that decision, the creator will be given a few weeks. If the asset can add "X" staking within that grace period, the challenge will resume again. (Loop after that)
- Users avoid assets that are at risk of being banned as attack assets.

**Comment by defi-er** [comment]
I will like to add some more context for the governance of assets. The first stage of onboarding is asset approval. We should have a transparent list of criteria for an asset to be able to onboard. The second stage is maintaining assets as some assets may become disqualified from the marketplace over time. For example, a crypto OSS platform eventually raises Venture Capital money in which case it doesn't need funding. In this scenario, governance should close the pool. 

btw- The solution's architecture you propose is identical to Kleros.

**Reply by aggre** [reply]
This idea technically aims to act as a disincentive for assets that the community considers "ineligible." (Therefore, eligibility wasn't the point I wanted to discuss. I should have shared that first)

I thought Kleros provides whitelists, but it also does seem to work as a "deletion decision." In essence, it doesn't matter whether Dev Protocol builds it in or use external protocols. I prefer a "report" mechanism rather than a "blocking at an entrance."

Dev Protocol deletes assets that are decided "delete" using external protocols such as Kleros. Attack assets rewards will be temporarily frozen as soon as they are reported. External protocols provide reporting incentives.
